addons/sale/controllers/portal.py

```diff
@@ -193,10 +208,20 @@ def _get_payment_values(self, order_sudo, downpayment=False, **kwargs):
         logged_in = not request.env.user._is_public()
         partner_sudo = request.env.user.partner_id if logged_in else order_sudo.partner_id
         company = order_sudo.company_id
-        if downpayment:
-            amount = order_sudo._get_prepayment_required_amount()
+        if not link_amount:
+            if downpayment:
+                amount = order_sudo._get_prepayment_required_amount() - order_sudo.amount_paid
+            else:
+                amount = order_sudo.amount_total - order_sudo.amount_paid
         else:
-            amount = order_sudo.amount_total - order_sudo.amount_paid
+            if installment == 'false':
+                if float(link_amount) >= order_sudo.amount_due:
```

Comment:
not sure if we should properly handle it but if float fails to convert `link_amount`   it will throw internal server error.

Reply:
I don't see a case of that happening other than someone putting it by themselves in link. As if there is no link_amount it will never get to that line.